### PR TITLE
Add ability to also recognize colons in ANSI escapes

### PR DESCRIPTION
**Pull request by jaysonlarose**
This basically just treats colons and semicolons as exactly the same when it comes to parsing ANSI.

Fixes: #166

### CODE REVIEW

**Comment by ssbarnea** [comment]
Can you please provide some backing for making this change? After 20+ years of using ANSI that is the first time when I hear about using colons. A quick search on wikipedia. There is a section on 24-bit which mentions as being included in ODA, a commercial failure.

https://en.wikipedia.org/wiki/ANSI_escape_code

Basically what bug does this fix?

**Comment by jaysonlarose** [comment]
See issue #166 for the full story.

**Review by ssbarnea — APPROVED** [review]

File: ansi2html/converter.py [hunk 1]
```diff
@@ -454,7 +454,11 @@ def _handle_ansi_code(
                 continue
 
             try:
-                params = list(map(int, params.split(";")))
+                params = [
+                    int(item)
+                    for subitem in [x.split(":") for x in params.split(";")]
+                    for item in subitem
+                ]
```
Comment (hartwork) [inline ansi2html/converter.py:461]: That seems to be way more complex than ideal and necessary. Would this approach work?:

```
In [7]: [int(v) for v in re.split("[:;]", "111;222:333;444")]
Out[7]: [111, 222, 333, 444]
```

File: ansi2html/converter.py [hunk 2]
```diff
@@ -325,7 +325,7 @@ def __init__(
             )
 
         self.vt100_box_codes_prog = re.compile("\033\\(([B0])")
-        self.ansi_codes_prog = re.compile("\033\\[" "([\\d;]*)" "([a-zA-z])")
+        self.ansi_codes_prog = re.compile("\033\\[" "([\\d;:]*)" "([a-zA-z])")
```
Comment (hartwork) [inline ansi2html/converter.py:328]: Before more research I'm not sure this can get us into a false positives scenario or not.

**Review by hartwork — CHANGES_REQUESTED** [review]

File: ansi2html/converter.py (hunk 1 above)
Comment (jaysonlarose) [reply ansi2html/converter.py:461]: Probably. I keep forgetting that `re.split()` exists.

File: ansi2html/converter.py (hunk 2 above)
Comment (hartwork) [reply ansi2html/converter.py:328]: I confirm now that colon is working in xterm as advertized.

File: ansi2html/converter.py (hunk 1 above)
Comment (hartwork) [reply ansi2html/converter.py:461]: @jaysonlarose if you could simplify the code using `re.split`, I would consider it ready to merge. Thank you!

Comment (jaysonlarose) [reply ansi2html/converter.py]: Done!

**Review by hartwork — APPROVED** [review]
@jaysonlarose thank you! :+1:

**Pull request merged**